pkg/controller.v1/common/scheduling.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2021 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	apiv1 "github.com/kubeflow/common/pkg/apis/common/v1"
+
+	log "github.com/sirupsen/logrus"
+	policyapi "k8s.io/api/policy/v1beta1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+type PGFillSpecFunc func(object metav1.Object) error
```

Comment:
Spell out PG?

Reply:
PodGroup. Maybe `FillPodGroupSpecFunc` will be more illustrative.